### DISCUSSION TITLE
Sporadic system crashes while sketchybar is running

### DISCUSSION THREAD

**Discussion by tanordheim** [FelixKratz/SketchyBar#400] [General]
Hi!

I'm experiencing some sporadic panics/crashes of macOS while sketchybar is running that I'm struggling to track down. I'm running the latest macOS Ventura (13.5.2), sketchybar v2.16.3. The system is also running yabai v5.0.8, with SIP disabled according to their docs. I wanted to post here instead of creating an issue, as I'm not confident sketchybar is the actual problem here, I am mostly looking for input/help on figuring out how to triage this.

These crashes happen almost daily, and often in conjunction with the machine going to sleep; its often rebooted after I come back from lunch for example. I have tried re-enabling SIP without that having any effect, I've tried the default/example sketchybar config from this repository, and so on but the crashes keeps on happening with the same frequency. Not running sketchybar removes those crashes completely, system goes from crashing daily to running stable for weeks. My current config + a captured panic log can be found at https://github.com/tanordheim/sketchybar-crash.

What makes me a bit uncertain on what the problem here is, is that the panic trace is a bit confusing - and doesn't mention sketchybar at all. When the crashes happen, its always core 0 crashing, and the task that panics is `logd`:

```
...
CORE 0 is the one that panicked. Check the full backtrace for details.
CORE 1: PC=0x00000001030088fc, LR=0x0000000102f3f528, FP=0x000000016da2c900
CORE 2: PC=0xfffffe0015a8abb0, LR=0xfffffe0015a8ab80, FP=0xfffffe881d603330
CORE 3: PC=0xfffffe0012dd2308, LR=0xfffffe0012dd2308, FP=0xfffffe881f4afef0
CORE 4: PC=0xfffffe0012dd2308, LR=0xfffffe0012dd2308, FP=0xfffffe881eca7ef0
CORE 5: PC=0xfffffe0012dd2308, LR=0xfffffe0012dd2308, FP=0xfffffe881e6dbef0
CORE 6: PC=0xfffffe0012dd2308, LR=0xfffffe0012dd2308, FP=0xfffffe881f43fef0
CORE 7: PC=0xfffffe0012dd2308, LR=0xfffffe0012dd2308, FP=0xfffffe881f0bbef0
CORE 8: PC=0xfffffe0012dd2308, LR=0xfffffe0012dd2308, FP=0xfffffe881eeffef0
CORE 9: PC=0xfffffe0012dd2308, LR=0xfffffe0012dd2308, FP=0xfffffe881f707ef0
Compressor Info: 0% of compressed pages limit (OK) and 4% of segments limit (OK) with 0 swapfiles and OK swap space
Panicked task 0xfffffe2cede14bb8: 3738 pages, 5 threads: pid 322: logd
Panicked thread: 0xfffffe2cec6ab010, backtrace: 0xfffffe8820216590, tid: 5644055
...
```

Has anybody else ever experienced this in any way? I couldn't find anything in the issue tracker, the discussions or on the internet in general about this which very strongly indicates it's a problem on my system and not with sketchybar in general. But the fact that these crashes only happen when sketchybar is running makes this a decent place to start digging I guess.

**Comment by FelixKratz** [comment]
That is very interesting. In principle no application with user permissions should be able to provoke such a crash, such that this could either point to a hardware problem or an actual security problem in apples os software. What you could try is a biscect of the commit history. I have recently changed some of the code interfacing with the os, so going back, lets say to v2.15.x, would be an interesting experiment.

I am not able to access to repo you have linked, is it private maybe?

**Reply by tanordheim** [reply]
Sorry, forgot to open the repo - should be accessible now :) 

As for bisecting I can give a downgrade a shot. This has been happening for me since I started using sketchybar in the start of August, presumably using 2.15.2. Would 2.15.1 or 2.14.4 be a good target to start out with?

**Reply by FelixKratz** [reply]
I have looked through the panic and the crash itself is not caused by sketchybar. Possibly running the bar puts the kernel in a more likely state to cause such a crash caused by a kernel bug or a hardware problem, however, I don‘t think it is the fault of the app. You could try not using external usb devices and displays for a while to see if this helps. This screams hardware problem to me.
